### ISSUE TITLE
Removing `PopupMenu` and `CanvasLayer` connected by `set_custom_viewport` crashes Godot

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#60052]
### Godot version

4.0.alpha.custom_build. ae2d53cdc

### System information

Ubuntu 21.10 - Nvidia GTX 970, Gnome shell 40.4 X11

### Issue description

When executing
```
func _process(delta):
	var temp_variable1763 = CanvasLayer.new()
	add_child(temp_variable1763)

	var temp_argument1763_f10_0 = PopupMenu.new()
	temp_variable1763.set_custom_viewport(temp_argument1763_f10_0)
	temp_argument1763_f10_0.queue_free()

	for i in get_children():
		i.queue_free()
```

Godot crashes with this backtrace
```
==24301==ERROR: AddressSanitizer: heap-use-after-free on address 0x61d000229290 at pc 0x000010936f3a bp 0x7ffcb8ac5b00 sp 0x7ffcb8ac5af0
READ of size 8 at 0x61d000229290 thread T0
    #0 0x10936f39 in CanvasLayer::_notification(int) scene/main/canvas_layer.cpp:178
    #1 0x1094a97d in CanvasLayer::_notificationv(int, bool) scene/main/canvas_layer.h:38
    #2 0x1a77bcf7 in Object::notification(int, bool) core/object/object.cpp:848
    #3 0x109b2463 in Node::_propagate_exit_tree() scene/main/node.cpp:292
    #4 0x109b1e54 in Node::_propagate_exit_tree() scene/main/node.cpp:283
    #5 0x109b1e54 in Node::_propagate_exit_tree() scene/main/node.cpp:283
    #6 0x10a00b68 in Node::_set_tree(SceneTree*) scene/main/node.cpp:2468
    #7 0x10ae58c2 in SceneTree::finalize() scene/main/scene_tree.cpp:564
    #8 0x23bfc18 in OS_LinuxBSD::run() platform/linuxbsd/os_linuxbsd.cpp:362
    #9 0x23acd6f in main platform/linuxbsd/godot_linuxbsd.cpp:68
    #10 0x7f4212981fcf in __libc_start_call_main ../sysdeps/nptl/libc_start_call_main.h:58
    #11 0x7f421298207c in __libc_start_main_impl ../csu/libc-start.c:409
    #12 0x23ac82d in _start (/usr/bin/godot4s+0x23ac82d)

0x61d000229290 is located 16 bytes inside of 2344-byte region [0x61d000229280,0x61d000229ba8)
freed by thread T0 here:
    #0 0x7f421395640f in __interceptor_free ../../../../src/libsanitizer/asan/asan_malloc_linux.cc:122
    #1 0x193353b6 in Memory::free_static(void*, bool) core/os/memory.cpp:168
    #2 0x256df5c in void memdelete<Object>(Object*) core/os/memory.h:111
    #3 0x10af37a2 in SceneTree::_flush_delete_queue() scene/main/scene_tree.cpp:1015
    #4 0x10ae2431 in SceneTree::process(double) scene/main/scene_tree.cpp:458
    #5 0x2556ec0 in Main::iteration() main/main.cpp:2722
    #6 0x23bf700 in OS_LinuxBSD::run() platform/linuxbsd/os_linuxbsd.cpp:357
    #7 0x23acd6f in main platform/linuxbsd/godot_linuxbsd.cpp:68
    #8 0x7f4212981fcf in __libc_start_call_main ../sysdeps/nptl/libc_start_call_main.h:58

previously allocated by thread T0 here:
    #0 0x7f4213956808 in __interceptor_malloc ../../../../src/libsanitizer/asan/asan_malloc_linux.cc:144
    #1 0x1933435b in Memory::alloc_static(unsigned long, bool) core/os/memory.cpp:75
    #2 0x1933426c in operator new(unsigned long, char const*) core/os/memory.cpp:40
    #3 0x1080ed40 in Object* ClassDB::creator<PopupMenu>() core/object/class_db.h:129
    #4 0x1a6f376f in ClassDB::instantiate(StringName const&) core/object/class_db.cpp:523
    #5 0x863d980 in GDScriptNativeClass::instantiate() modules/gdscript/gdscript.cpp:92
    #6 0x863d361 in GDScriptNativeClass::_new() modules/gdscript/gdscript.cpp:80
    #7 0x8723a79 in void call_with_variant_args_ret_helper<__UnexistingClass, Variant>(__UnexistingClass*, Variant (__UnexistingClass::*)(), Variant const**, Variant&, Callable::CallError&, IndexSequence<>) core/variant/binder_common.h:654
    #8 0x87234d4 in void call_with_variant_args_ret_dv<__UnexistingClass, Variant>(__UnexistingClass*, Variant (__UnexistingClass::*)(), Variant const**, int, Variant&, Callable::CallError&, Vector<Variant> const&) core/variant/binder_common.h:467
    #9 0x8722bae in MethodBindTR<Variant>::call(Object*, Variant const**, int, Callable::CallError&) core/object/method_bind.h:495
    #10 0x8bd765b in GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Callable::CallError&, GDScriptFunction::CallState*) modules/gdscript/gdscript_vm.cpp:1633
    #11 0x8674ef5 in GDScriptInstance::callp(StringName const&, Variant const**, int, Callable::CallError&) modules/gdscript/gdscript.cpp:1539
    #12 0x10a39925 in bool Node::_gdvirtual__process_call<false>(double) scene/main/node.h:229
    #13 0x109a4f44 in Node::_notification(int) scene/main/node.cpp:56
    #14 0x3057457 in Node::_notificationv(int, bool) scene/main/node.h:45
    #15 0x1a77bcf7 in Object::notification(int, bool) core/object/object.cpp:848
    #16 0x10aef3b5 in SceneTree::_notify_group_pause(StringName const&, int) scene/main/scene_tree.cpp:851
    #17 0x10ae1ea5 in SceneTree::process(double) scene/main/scene_tree.cpp:450
    #18 0x2556ec0 in Main::iteration() main/main.cpp:2722
    #19 0x23bf700 in OS_LinuxBSD::run() platform/linuxbsd/os_linuxbsd.cpp:357
    #20 0x23acd6f in main platform/linuxbsd/godot_linuxbsd.cpp:68
    #21 0x7f4212981fcf in __libc_start_call_main ../sysdeps/nptl/libc_start_call_main.h:58

```

This example was found by Godot fuzzer - [Qarminer](https://github.com/qarmin/Qarminer), so it is quite unlikelly that this code could be used in real project, but still this should be handled gracefully.

### Steps to reproduce

Above

**Comment by qarmin** [comment]
This is very similar issue to https://github.com/godotengine/godot/issues/53445, so maybe fix will be also similar

**Comment by akien-mga** [comment]
Seems fixed in 4.1.

Spams this error, but no (immediate) crash:
```
ERROR: Cannot disconnect 'child_order_changed' from callable 'Viewport::canvas_parent_mark_dirty': the callable object is null.
   at: _disconnect (./core/object/object.cpp:1343)
ERROR: Condition "!viewport" is true.
   at: viewport_remove_canvas (./servers/rendering/renderer_viewport.cpp:1074)
```
